diff --git a/ffmpeg.c b/ffmpeg.c
index b26995deb0..c8839fcc3f 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -1669,7 +1669,7 @@ static void print_report(int is_last_report, int64_t timer_start, int64_t cur_ti
     }
 
     if (print_stats || is_last_report) {
-        const char end = is_last_report ? '\n' : '\r';
+        const char end = '\n';
         if (print_stats==1 && AV_LOG_INFO > av_log_get_level()) {
             fprintf(stderr, "%s    %c", buf, end);
         } else
